server/plugin/plugin.go

```diff
@@ -250,6 +274,151 @@ func (p *Plugin) OnDeactivate() error {
 	return nil
 }
 
+func (p *Plugin) getPostPropsForReaction(reaction *model.Reaction) (orgRepo []string, id float64, objectType string, ok bool) {
+	post, err := p.client.Post.GetPost(reaction.PostId)
+	if err != nil {
+		p.API.LogDebug("Error fetching post for reaction", "error", err.Error())
+		return orgRepo, id, objectType, false
+	}
+
+	repo, ok := post.GetProp(postPropGithubRepo).(string)
+	if !ok || repo == "" {
+		return orgRepo, id, objectType, false
+	}
+
+	orgRepo = strings.Split(repo, "/")
+	if len(orgRepo) != 2 {
+		p.API.LogDebug("Invalid organization repository")
+		return orgRepo, id, objectType, false
+	}
+
+	id, ok = post.GetProp(postPropGithubObjectID).(float64)
+	if !ok || id == 0 {
+		return orgRepo, id, objectType, false
+	}
+
+	objectType, ok = post.GetProp(postPropGithubObjectType).(string)
+	if !ok || objectType == "" {
+		return orgRepo, id, objectType, false
+	}
+
+	return orgRepo, id, objectType, true
+}
+
+func (p *Plugin) ReactionHasBeenAdded(c *plugin.Context, reaction *model.Reaction) {
+	githubEmoji := p.emojiMap[reaction.EmojiName]
+	if githubEmoji == "" {
+		return
+	}
+
+	orgRepo, id, objectType, ok := p.getPostPropsForReaction(reaction)
+	if !ok {
+		return
+	}
+
+	info, appErr := p.getGitHubUserInfo(reaction.UserId)
+	if appErr != nil {
+		if appErr.ID != apiErrorIDNotConnected {
+			p.API.LogDebug("Error in getting user info", "error", appErr.Message)
+		}
+		return
+	}
+
+	ghClient := p.githubConnectUser(context.Background(), info)
+	owner, repo := orgRepo[0], orgRepo[1]
+	switch objectType {
+	case githubObjectTypeIssueComment:
+		if _, _, err := ghClient.Reactions.CreateIssueCommentReaction(context.Background(), owner, repo, int64(id), githubEmoji); err != nil {
+			p.API.LogDebug("Error occurred while creating issue comment reaction", "error", err.Error())
+			return
+		}
+	case githubObjectTypeIssue:
+		if _, _, err := ghClient.Reactions.CreateIssueReaction(context.Background(), owner, repo, int(id), githubEmoji); err != nil {
+			p.API.LogDebug("Error occurred while creating issue reaction", "error", err.Error())
+			return
+		}
+	case githubObjectTypePRReviewComment:
+		if _, _, err := ghClient.Reactions.CreatePullRequestCommentReaction(context.Background(), owner, repo, int64(id), githubEmoji); err != nil {
+			p.API.LogDebug("Error occurred while creating PR review comment reaction", "error", err.Error())
+			return
+		}
+	default:
+		return
+	}
+}
+
+func (p *Plugin) ReactionHasBeenRemoved(c *plugin.Context, reaction *model.Reaction) {
+	githubEmoji := p.emojiMap[reaction.EmojiName]
+	if githubEmoji == "" {
+		return
```

Comment:
logging info or warning here to know why the emoji is not in the list (with reaction.EmojiName to show on the log)?